### PR TITLE
Workspaces supports connecting to a server with different accounts.

### LINKED ISSUE #2977: Workspaces should allow user to connect with different accounts.

### PR DESCRIPTION
**Pull request by karthiknadig**
fixes #2977

### CODE REVIEW

**Comment by karthiknadig** [comment]
This PR changes the way we store credentials, from server url as key to workspace name as key.
image

**Comment by int19h** [comment]
👍

File: src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs [hunk 1]
```diff
@@ -133,13 +134,7 @@ internal class ConnectionManager : IConnectionManager {
...
             if (isRemoved) {
                 UpdateRecentConnections();
-
-                // Credentials are saved by URI. Delete the credentials if there are no other connections using it.
-                if (_connections.All(kvp => kvp.Value.Uri != connection.Uri)) {
-                    if (connection.Uri != null) {
-                        _securityService.DeleteUserCredentials(connection.Uri.ToCredentialAuthority());
-                    }
-                }
+                _securityService.DeleteUserCredentials(connection.ToCredentialAuthority());
```
Comment (AlexanderSher) [inline src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs]: `connection.BrokerConnectionInfo.CredentialAuthority`

And you can remove `ToCredentialAuthority` with its class.

File: src/Host/Client/Impl/Host/BrokerConnectionInfo.cs [hunk 2]
```diff
@@ -4,18 +4,21 @@
 using System;
 using System.Diagnostics;
 using Microsoft.Common.Core;
+using Microsoft.Common.Core.Security;
 
 namespace Microsoft.R.Host.Client.Host {
     [DebuggerDisplay("{Uri}, IsRemote={IsRemote}, InterpreterId={InterpreterId}")]
     public struct BrokerConnectionInfo {
+        public string Name { get; }
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/Host/BrokerConnectionInfo.cs]: Do you use this property anywhere outside this class, or is it just required for `CredentialAuthority`?

Comment (karthiknadig) [inline src/Host/Client/Impl/Host/BrokerConnectionInfo.cs]: It is used in `RemoteBrokerClient `while creating `RemoteCredentialDecorator`

**Pull request merged**